### PR TITLE
helm: make eg-addons support IPv6 cluster

### PR DESCRIPTION
**Pull request by zirain**
xref: https://github.com/envoyproxy/gateway/pull/4550#issuecomment-2442961025

### CODE REVIEW

**Comment by zirain** [comment]
test on my linux(IPv6 kind cluster) and macos m1(IPv4 kind cluster), full test will be https://github.com/envoyproxy/gateway/pull/4550

File: internal/gatewayapi/testdata/securitypolicy-with-jwt-and-invalid-oidc.out.yaml [hunk 1]
```diff
@@ -173,7 +173,8 @@ securityPolicies:
         sectionName: http
       conditions:
       - lastTransitionTime: null
-        message: 'OIDC: secret default/client2-secret does not exist.'
+        message: 'OIDC: error fetching endpoints from issuer: Get "https://accounts.google.com/.well-known/openid-configuration":
```
Comment (zhaohuabing) [inline internal/gatewayapi/testdata/securitypolicy-with-jwt-and-invalid-oidc.out.yaml:176]: :-)

Comment (zirain) [reply internal/gatewayapi/testdata/securitypolicy-with-jwt-and-invalid-oidc.out.yaml]: nit

File: .gitignore [hunk 2]
```diff
@@ -34,6 +34,7 @@ charts/gateway-helm/values.yaml
 
 # dependency charts generated by addons helm.
 charts/gateway-addons-helm/charts/
+charts/gateway-addons-helm/tmpcharts/
```
Comment (zhaohuabing) [inline .gitignore:37]: Do we need this?

File: charts/gateway-addons-helm/values.yaml [hunk 3]
```diff
@@ -186,7 +186,7 @@ opentelemetry-collector:
   config:
     exporters:
       prometheus:
-        endpoint: 0.0.0.0:19001
+        endpoint: "[${env:MY_POD_IP}]:19001"
```
Comment (zhaohuabing) [inline charts/gateway-addons-helm/values.yaml:189]: Nit: does ::19001 work?

File: .gitignore (hunk 2 above)
Comment (zirain) [reply .gitignore:37]: better to keep

File: charts/gateway-addons-helm/values.yaml (hunk 3 above)
Comment (zirain) [reply charts/gateway-addons-helm/values.yaml:189]: no, it would pass the internal check.

Comment (zirain) [reply charts/gateway-addons-helm/values.yaml:189]:
```
_, _, err := net.SplitHostPort("::19001")
	if err != nil {
		fmt.Printf("err: %v", err)
	}
```
```
err: address ::19001: too many colons in address
```

File: .gitignore (hunk 2 above)
Comment (zhaohuabing) [reply .gitignore:37]: Curious why this is needed?

Comment (zirain) [reply .gitignore:37]: there's some tmp files, make sure noone will commit it by accident.

File: charts/gateway-addons-helm/values.yaml [hunk 4]
```diff
@@ -209,17 +207,36 @@ opentelemetry-collector:
...
+          scrape_configs:
+          - job_name: opentelemetry-collector
+            scrape_interval: 10s
+            static_configs:
+            - targets:
+              - "[${env:MY_POD_IP}]:8888"
     service:
+      telemetry:
+        metrics:
+          address: "[${env:MY_POD_IP}]:8888"
```
Comment (arkodg) [inline charts/gateway-addons-helm/values.yaml:239]: who is populating this env ?

Comment (zirain) [reply charts/gateway-addons-helm/values.yaml:239]: otel-collector pod

**Review by arkodg — APPROVED** [review]
LGTM thanks !
cc @juwon8891

File: charts/gateway-addons-helm/values.yaml (hunk 4 above)
Comment (zirain) [reply charts/gateway-addons-helm/values.yaml:239]: we'd better change upstream charts, I do believe they should change the default value.

**Review by zhaohuabing — APPROVED** [review]

**Pull request merged**